validate/validate.go

```diff
@@ -114,6 +118,49 @@ func (v *Validator) CheckAll() error {
 	return errs.ErrorOrNil()
 }
 
+// JSONSchemaURL returns the URL for the JSON Schema specifying the
+// configuration format.  It consumes configSchemaTemplate, but we
+// provide it as a function to isolate consumers from inconsistent
+// naming as runtime-spec evolves.
+func JSONSchemaURL(version string) (url string, err error) {
+	ver, err := semver.Parse(version)
+	if err != nil {
+		return "", err
+	}
+	configSchemaVersion, err := semver.Parse("1.0.0-rc2") // config.json became config-schema.json in 1.0.0-rc2
```

Comment:
should be 1.0.0?

Reply:
> should be 1.0.0?

No, rc2 is when `config.jdon` became `config-schema.json`.  `JSONSchemaURL` works with all of our releases, not just `1.0.0`.

Reply:
Sorry, I can't understand.
Our code now works based on runtime-spec v1.0.0.
Even though schema changes begin rc2,  why must be rc2?  It's not the latest released version.

Reply:
> Even though schema changes begin rc2, why must be rc2? It's not the latest released version.

I think the long-term goal is to be able to test against multiple spec versions.  See #98 for more on this.  @liangchenye at least seemed on board with that idea (if not with my Python approach to it) based on his comment [here][1].  I've pushed ea82fad → 9d07098 showing a example of using `CheckJSONSchema` on an older release (although I didn't go back before rc2) in case that helps clarify.  I understand that the rest of the Go-based validation is specific to 1.0.0, but don't think that is a reason to force `CheckJSONSchema` to be specific to 1.0.0.  Having a flexible `CheckJSONSchema` means we'll have less work to do when we eventually adjust `validate` to work with multiple versions.

The flexibility of `CheckJSONSchema` is somewhat limited in the current implementation because it happens after we read JSON into Go (and in some `validate` workflows there may never have been JSON to begin with).  That means we miss things like:

```json
{
  "ociVersion": "1.0.0-rc6",
  "linux": {
    "resources": {
      "disableOOMKiller": "I should be a boolean"
    }
  }
}
```

because since opencontainers/runtime-spec#896 there has been no way to represent `linux.resources.disableOOMKiller` in Go.  But I expect folks will mostly be interested in 1.0.0+, and the SemVer commitment will help us avoid that sort of issue for the 1.x line.

[1]: https://github.com/opencontainers/runtime-tools/issues/98#issuecomment-223878770

Reply:
Ah,  sorry.  I misunderstand the purpose of configSchemaVersion.

Reply:
> I misunderstand the purpose of configSchemaVersion.

Maybe I should rename it to `configRenamedToConfigSchemaVersion` or some such?

Reply:
I think it will be better

Reply:
> I think it will be better

Renamed in 9d07098 → 3935592.